build/src/lib.rs

```diff
@@ -227,21 +367,25 @@ impl BinaryBuilder {
 
         self.outputs.push(tokenizer_out);
         self.outputs.push(rules_out);
+        Ok(())
     }
 
     /// Creates a new binary builder. `language_codes` must be in ISO 639-1 (two-letter) format.
-    /// If `language_codes` is `None`, uses all supported languages.
+    /// If `language_codes` is `&[]`, uses all supported languages.
     /// If this is used in a `build.rs`, `out_dir` should probably be the OUT_DIR environment variable.
-    pub fn new<P: AsRef<Path>>(language_codes: Option<&[&str]>, out_dir: P) -> Self {
-        let language_codes: Vec<_> = language_codes.map_or_else(
-            || {
-                supported_language_codes()
-                    .into_iter()
-                    .map(|x| x.to_owned())
-                    .collect()
-            },
-            |x| x.iter().map(|x| x.to_string()).collect(),
-        );
+    pub fn new<P: AsRef<Path>>(language_codes: &[&str], out_dir: P) -> Self {
+        let language_codes: Vec<_> = if language_codes.is_empty() {
+            supported_language_codes()
+                .into_iter()
+                .map(ToOwned::to_owned)
+                .collect()
+        } else {
+            language_codes
+                .into_iter()
```

Comment:
Locally I get a warning that this `into_iter` is equivalent to `iter` but somehow CI clippy doesn't warn about it.

Reply:
For slices this is indeed the case. Addressing.